integration-cli/docker_cli_daemon_test.go

```diff
@@ -2136,14 +2137,14 @@ func (s *DockerDaemonSuite) TestDaemonStartWithoutColors(c *testing.T) {
 
 	go func() {
 		io.Copy(b, p)
-		done <- true
```

Comment:
These changes just seem to be reversing what's happening here.
Is there something else happening that I'm missing?

The intent of this is to signal to the main test that the copy is done hence the send in the goroutine.
The change seems to be flipping that to have the main test effectively ask "are you done" which I find to be a little more confusing.

Reply:
I found a way to solve this problem. I switched send and receive of `done` back to old version (before I open this PR), and add 1 buffer to `done`. In this way, `done <- true` won't block even if main goroutine exits early. This also doesn't change the semantic of the program: `done <- true` is the last instruction of child goroutine, so whether it blocks or not doesn't matter, and `<-done` in mian goroutine will still block.